vcd/resource_vcd_vapp_network.go

```diff
@@ -58,11 +58,22 @@ func resourceVcdVappNetwork() *schema.Resource {
 				Description: "Optional description for the network",
 			},
 			"netmask": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				ForceNew:    true,
-				Default:     "255.255.255.0",
-				Description: "Netmask address for a subnet. Default is 255.255.255.0",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				Deprecated:   "Use prefix_length instead which supports both IPv4 and IPv6",
+				Description:  "Netmask address for a subnet.",
+				ExactlyOneOf: []string{"prefix_length"},
+			},
+			"prefix_length": {
+				Type:         schema.TypeString,
```

Comment:
This one has skipped my attention when I reviewed [PR 550](https://github.com/vmware/go-vcloud-director/pull/550/files). Why is this field a string and not an Int?
According to [the API](https://developer.vmware.com/apis/1507/vmware-cloud-director/doc/doc/types/IpScopeType.html), it should be an integer.

Reply:
Mostly because `TypeInt` defaults to `0` and `TypeString` defaults to `""` and there could be an edge case where one creates a network with `prefix_length=0`, but also if someone provides a netmask, it seems wrong to have checks whether `netmask` is given, to then nullify the `prefix_length`. 

Took example from this code: https://github.com/vmware/terraform-provider-vcd/blob/main/vcd/resource_vcd_nsxt_alb_edgegateway_service_engine_group.go#L65-L73
cc @Didainius